js/switcheo.js

```diff
@@ -0,0 +1,370 @@
+        let signedMessage = this.signMessageHash (hexMessage, this.privateKey);
+        createOrder['signature'] = signedMessage;
+        createOrder['address'] = this.walletAddress.toLowerCase ();
+        let headers = {
+            'Content-type': 'application/json',
+            'Accept': 'text/plain',
+        };
+        let submitCreateOrder = await this.privatePostOrders (params, headers, createOrder);
+        let orderId = submitCreateOrder['id'];
+        let signedCreateOrder = this.signCreateOrder (submitCreateOrder, this.privateKey);
+        let submitExecuteOrder = await this.privatePostOrdersIdBroadcast ({ 'id': orderId }, headers, signedCreateOrder);
+        return submitExecuteOrder;
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        let cancelRequest = {
+            'order_id': id,
+            'timestamp': this.milliseconds (),
+        };
+        let stableStringify = this.stringifyMessage (cancelRequest);
+        let hexMessage = this.toHex (stableStringify);
+        let signedMessage = this.signMessageHash (hexMessage, this.privateKey);
+        cancelRequest['signature'] = signedMessage;
+        cancelRequest['address'] = this.walletAddress.toLowerCase ();
+        let headers = {
+            'Content-type': 'application/json',
+            'Accept': 'text/plain',
+        };
+        let createCancel = await this.privatePostCancellations (params, headers, cancelRequest);
+        let cancelId = createCancel['id'];
+        let signedCancel = this.signMessageHash (createCancel['transaction']['sha256'], this.privateKey);
+        let executeCancel = await this.privatePostCancellationsIdBroadcast ({ 'id': cancelId }, headers, { 'signature': '0x' + signedCancel });
+        return executeCancel;
+    }
+
+    async deposit (symbol, amount, params = {}) {
+        let depositRequest = {
+            'blockchain': 'eth',
+            'asset_id': symbol,
+            'amount': this.toWei (amount, 'ether'),
+            'contract_hash': this.fetchContractHash ()['ETH'],
+            'timestamp': this.milliseconds (),
+        };
+        let stableStringify = this.stringifyMessage (depositRequest);
+        let hexMessage = this.toHex (stableStringify);
+        let signedMessage = this.signMessageHash (hexMessage, this.privateKey);
+        depositRequest['signature'] = signedMessage;
+        depositRequest['address'] = this.walletAddress.toLowerCase ();
+        let headers = {
+            'Content-type': 'application/json',
+            'Accept': 'text/plain',
+        };
+        let createDeposit = await this.privatePostDeposits (params, headers, depositRequest);
+        let depositId = createDeposit['id'];
+        let executeDeposit = await this.privatePostDepositsIdBroadcast ({ 'id': depositId }, headers, this.signTransaction (createDeposit, this.privateKey));
+        return executeDeposit;
+    }
+
+    async withdrawal (symbol, amount, params = {}) {
```

Comment:
The unified API doesn't have a `withdrawal` method, but has a `withdraw` method, and it has a certain signature that we have to comply with: https://github.com/ccxt/ccxt/wiki/Manual#withdraw

Reply:
Changed function name to `withdraw` and ensured it complied with the unified signature.

This commit should reflect the changes: https://github.com/ccxt/ccxt/pull/4822/commits/29c374af2e15303bd84ba941a03197b6ade84859